http3/server.go

```diff
@@ -492,15 +530,32 @@ func (s *Server) handleConn(conn quic.Connection) error {
 	// Process all requests immediately.
 	// It's the client's responsibility to decide which requests are eligible for 0-RTT.
 	for {
-		str, datagrams, err := hconn.acceptStream(context.Background())
+		str, datagrams, err := hconn.acceptStream(s.closeCtx)
 		if err != nil {
+			// server is closed, send GOAWAY frame and return, since those requests will be rejected anyway
+			if s.closeCtx.Err() != nil {
+				b = (&goawayFrame{
+					StreamID: lastID + 4,
+				}).Append(b[:0])
+				// set a deadline to send the GOAWAY frame
+				ctrlStr.SetWriteDeadline(time.Now().Add(5 * time.Second))
```

Comment:
Nit: Make this a constant at the beginning of the file? Makes it look less like a magic value.